handlers/handlers.go

```diff
@@ -0,0 +1,369 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"gin/models"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/olivere/elastic/v7"
+	"gopkg.in/redis.v5"
+	"html"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Server struct {
+	ElasticClient *elastic.Client
+	RedisClient   *redis.Client
+}
+
+func (server *Server) validateSearchBook(title string, titleOk bool, authorName string, authorNameOk bool, price string, priceOk bool, ebookAvailable string, ebookAvailableOk bool, publishDate string, publishDateOk bool) error {
+	if (!titleOk || title == "") && (!authorNameOk || authorName == "") && (!priceOk || price == "") && (!ebookAvailableOk || ebookAvailable == "") && (!publishDateOk || publishDate == "") {
+		return fmt.Errorf("no parameter given. requires at least one of the following: 'title', 'authorName', 'price', 'ebookAvailable', 'publishDate")
+	}
+	return nil
+}
+
+func (server *Server) validateCreateBook(title, authorName, price, ebookAvailable, publishDate string) error {
+	missingParameter := make([]string, 0)
+
+	if title == "" {
+		missingParameter = append(missingParameter, "'title'")
+	}
+	if authorName == "" {
+		missingParameter = append(missingParameter, "'authorName'")
+	}
+	if price == "" {
+		missingParameter = append(missingParameter, "'price'")
+	}
+	if ebookAvailable == "" {
+		missingParameter = append(missingParameter, "'ebookAvailable'")
+	}
+	if publishDate == "" {
+		missingParameter = append(missingParameter, "'publishDate'")
+	}
+
+	if len(missingParameter) != 0 {
+		return fmt.Errorf("the following parameters are missing: %v", strings.Join(missingParameter, ","))
+	}
+
+	return nil
+}
+
+func (server *Server) CreateBook(c *gin.Context) {
+	title := c.PostForm("title")
+	authorName := c.PostForm("authorName")
```

Comment:
url fields should be named with under_score, Go names with camelCase.
```suggestion
	authorName := c.PostForm("author_name")
```